shell/platform/windows/testing/windows_test_config_builder.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_TESTING_WINDOWS_TEST_CONFIG_BUILDER_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_TESTING_WINDOWS_TEST_CONFIG_BUILDER_H_
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/unique_object.h"
+#include "flutter/shell/platform/windows/public/flutter_windows.h"
+#include "flutter/shell/platform/windows/testing/windows_test_context.h"
+
+namespace flutter {
+namespace testing {
+
+// Deleter for FlutterDesktopEngineRef objects.
+struct EngineDeleter {
+  typedef FlutterDesktopEngineRef pointer;
+  void operator()(FlutterDesktopEngineRef engine) {
+    FML_CHECK(FlutterDesktopEngineDestroy(engine));
+  }
+};
+
+// Unique pointer wrapper for FlutterDesktopEngineRef.
+using EnginePtr = std::unique_ptr<FlutterDesktopEngine, EngineDeleter>;
+
+// Deleter for FlutterViewControllerRef objects.
+struct ViewControllerDeleter {
+  typedef FlutterDesktopViewControllerRef pointer;
+  void operator()(FlutterDesktopViewControllerRef engine) {
+    FlutterDesktopViewControllerDestroy(engine);
```

Comment:
Note that this doesn't return a boolean which is why there's no `FML_CHECK` here.